src/CauserResolver.php

```diff
@@ -0,0 +1,115 @@
+<?php
+namespace Spatie\Activitylog;
+
+use Closure;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Database\Eloquent\Model;
+use InvalidArgumentException;
+use Spatie\Activitylog\Exceptions\CouldNotLogActivity;
+
+class CauserResolver
+{
+    protected AuthManager $authManager;
+
+    protected string $authDriver;
+
+    /**
+     * User defined model or callback to override default reslover logic
+     */
+    protected Model | Closure | null $override = null;
+
+    public function __construct()
+    {
+        $config = app(Repository::class);
+        $this->authManager = app(AuthManager::class);
```

Comment:
This class is bonded and resolved using the container so we can use dependency injection.

```suggestion
    public function __construct(Repository $config, AuthManager $authManager)
    {
        $this->authManager = $authManager;
```